### PR TITLE
Node 18+ updates #6

### PR DESCRIPTION
**Pull request by fajar-apri-alaska**
# Alaska Airlines Pull Request

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

**Resolves:** #56, #57, #58, #63 

## Summary:

...

## Type of change:

Please delete options that are not relevant.

- [x] Revision of an existing capability

## Checklist:

- [x] My update follows the CONTRIBUTING guidelines of this project
- [x] I have performed a self-review of my own update

**By submitting this Pull Request, I confirm that my contribution is made under the terms of the Apache 2.0 license.**

_Pull Requests will be evaluated by their quality of update and whether it is consistent with the goals and values of this project. Any submission is to be considered a conversation between the submitter and the maintainers of this project and may require changes to your submission._

**Thank you for your submission!**<br>
-- Auro Design System Team

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -15,20 +15,20 @@ The following sections are editable by making changes to the following files:
 | Use Cases              | Examples for when to use this component           | `./docs/partials/useCases.md`       |
 | Additional Information | For use to add any component specific information | `./docs/partials/readmeAddlInfo.md` |
 | Component Example Code | HTML sample code of the components use            | `./apiExamples/basic.html`          |
--->
-
+-->
+
```
Comment (blackfalcon) [inline README.md]: This file should not be updated directly. It is generated with the `npm run build:docs` npm script. 

All updates need to be either addressed in the `./docTemplates/README.md` file `./docs/partials` directory or updated in the Auro Generator that @irma-kurnia-alaska is making.

File: demo/index.html [hunk 2]
```diff
@@ -4,33 +4,26 @@
...
+    <link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/@alaskaairux/design-tokens@latest/dist/tokens/CSSCustomProperties.css">
+    <link rel="stylesheet" type="text/css" href="https://cdn.jsdelivr.net/npm/@aurodesignsystem/webcorestylesheets@latest/dist/demoWrapper.css" />
+    <link rel="stylesheet" type="text/css" href="https://cdn.jsdelivr.net/npm/@aurodesignsystem/webcorestylesheets@latest/dist/elementDemoStyles.css" />
   </head>
   <body>
     <main></main>
 
     <script type="module">
-      import 'https://unpkg.com/marked@latest/marked.min.js';
-      import 'https://unpkg.com/prismjs@latest/prism.js';
```
Comment (blackfalcon) [inline demo/index.html:21]: Why was this deleted? It should be updated.

**Review by blackfalcon — CHANGES_REQUESTED** [review]
Please see this commit for all changes that should be related to this issue 

- https://github.com/AlaskaAirlines/auro-button/pull/210/commits/e10e5135ea2133796a4b706e3476a8a31dc5c91d

File: README.md (hunk 1 above)
Comment (fajar-apri-alaska) [reply README.md]: I think I did try the npm run build first before git commit to make sure I'm not breaking something.
I will revert back the changes for this file only then.

File: demo/index.html (hunk 2 above)
Comment (fajar-apri-alaska) [reply demo/index.html:21]: On it. But going to update it to cdn link.

File: .github/ISSUE_TEMPLATE/bug_report.yml [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+    attributes:
+      value: >
+        [![See it on
+        NPM!](https://img.shields.io/npm/v/@alaskaairux/auro-flight?style=for-the-badge&color=orange)](https://www.npmjs.com/package/@alaskaairux/auro-flight)
+  - type: textarea
+    id: details
+    attributes:
+      label: Please describe the bug
+      description: List out the steps to reproduce the behavior
+      placeholder: 1. Go to '...' 2. Click on '....' 3. Scroll down to '....' 4. See error
```
Comment (blackfalcon) [inline .github/ISSUE_TEMPLATE/bug_report.yml:27]:
```suggestion
    placeholder: |
        1. Go to '...'
        2. Click on '....'
        3. Scroll down to '....'
        4. See error
```

File: .github/ISSUE_TEMPLATE/general-support.yml [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+  - type: textarea
+    id: solution
+    attributes:
+      label: Possible Solution
+      description: Not obligatory, but suggest an idea of how to implement the requested update.
+  - type: textarea
+    id: context
+    attributes:
+      label: Additional context
+      description: Add any other context or screenshots about the feature request here.
```
Comment (blackfalcon) [inline .github/ISSUE_TEMPLATE/general-support.yml:24]:
```suggestion
  - type: textarea
    id: context
    attributes:
      label: Additional context
      description: Add any other context or screenshots about the feature request here.
  - type: textarea
    id: exit-criteria
    attributes:
      label: Exit criteria
      description: Define your issue's exit criteria
      placeholder: "This issue can be closed once the error has been corrected. Any additional scope e.g. updates or refactors, should be discussed PRIOR to adding scope to this issue."

```

File: .github/settings.yml [hunk 5]
```diff
@@ -71,59 +72,101 @@ branches:
...
+    - beginner
+    - good-starter-issue
+    - good for beginner
+    - Good for beginners
+    - starter-issue
+    - starter
+  - name: help wanted
+    color: '6bb7fb'
+    description: Extra attention is needed, this user requires assistance to complete
+      the work
```
Comment (blackfalcon) [inline .github/settings.yml:95]:
```suggestion
  - name: duplicate
    color: '156fad'
    description: This Issue or Pull Request already exists
  - name: not-reviewed
    color: '222222'
    description: Issue has not been reviewed by Auro team members
  - name: good first issue
    color: '6bb7fb'
    description: Good for newcomers
    aliases:
    - beginner-friendly
    - beginner
    - good-starter-issue
    - good for beginner
    - Good for beginners
    - starter-issue
    - starter
  - name: help wanted
    color: '6bb7fb'
    description: Extra attention is needed, this user requires assistance to complete
      the work
```

File: demo/index.html [hunk 6]
```diff
@@ -28,9 +29,14 @@
...
     </script>
     <script type="module" src="../src/auro-flight.js"></script>
-    <script src="https://unpkg.com/@alaskaairux/auro-accordion@latest/dist/auro-accordion__bundled.js" type="module"></script>
-    <script src="https://unpkg.com/@alaskaairux/auro-icon@latest/dist/auro-icon__bundled.js" type="module"></script>
-    <script src="https://unpkg.com/@alaskaairux/auro-alerts@latest/dist/auro-alerts__bundled.js" type="module"></script>
-    <script src="https://unpkg.com/@alaskaairux/auro-datetime@latest/dist/auro-datetime__bundled.js" type="module"></script>
+    <script type="module">
+      import { registerComponent } from "../index.js"
+      registerComponent('custom-flight');
+    </script>
```
Comment (blackfalcon) [inline demo/index.html:35]: This is not needed?

File: index.js [hunk 7]
```diff
@@ -1 +1,15 @@
...
+ * @returns {void}
+ */
+ const registerComponent = (name = 'custom-flight') => {
+  // alias definition
+  if (!customElements.get(name)) {
+    customElements.define(name, class extends AuroFlight {});
+  }
+}
+
+export { registerComponent }
```
Comment (blackfalcon) [inline index.js:15]: Please set your IDE to always add a line after.

File: package.json [hunk 8]
```diff
@@ -16,71 +16,69 @@
...
-    "@web/dev-server": "^0.1.32",
-    "@web/test-runner": "^0.13.31",
-    "autoprefixer": "^10.4.7",
-    "babel-loader": "^8.2.5",
+    "@semantic-release/npm": "^10.0.3",
+    "@types/estree": "^1.0.1",
+    "@web/dev-server": "^0.2.1",
+    "@web/test-runner": "^0.16.1",
+    "autoprefixer": "^10.4.14",
+    "babel-loader": "^9.1.2",
```
Comment (blackfalcon) [inline package.json:49]: Remove

File: package.json [hunk 9]
```diff
@@ -16,71 +16,69 @@
...
+    "@alaskaairux/auro-flightline": "^1.4.0",
     "chalk": "^4.1.2",
-    "lit-element": "^2.5.1",
-    "lit-html": "^1.4.1"
+    "lit": "^2.7.3"
   },
   "peerDependencies": {
     "@alaskaairux/auro-datetime": "^1.3.0",
     "@alaskaairux/design-tokens": "^3.9.1",
     "@alaskaairux/webcorestylesheets": "^3.7.3"
```
Comment (blackfalcon) [inline package.json:30]:
```suggestion
    "@aurodesignsystem/webcorestylesheets": "^4.3.1"
```

File: package.json [hunk 10]
```diff
@@ -16,71 +16,69 @@
...
   },
   "devDependencies": {
-    "@alaskaairux/auro-datetime": "^1.3.0",
-    "@alaskaairux/design-tokens": "^3.11.0",
-    "@alaskaairux/icons": "^4.14.0",
+    "@alaskaairux/auro-datetime": "^1.4.0",
+    "@alaskaairux/design-tokens": "^3.13.0",
+    "@alaskaairux/icons": "^4.26.0",
     "@alaskaairux/webcorestylesheets": "^3.7.3",
```
Comment (blackfalcon) [inline package.json:36]:
```suggestion
     "@aurodesignsystem/webcorestylesheets": "^4.3.1",
```

File: package.json [hunk 11]
```diff
@@ -16,71 +16,69 @@
...
+    "core-js": "^3.30.1",
+    "eslint": "^8.39.0",
+    "eslint-plugin-jsdoc": "^43.1.1",
+    "husky": "^8.0.3",
     "lodash": "^4.17.21",
-    "markdown-magic": "^2.6.0",
-    "marked": "^4.0.17",
-    "nodemon": "^2.0.18",
+    "markdown-magic": "^2.6.1",
+    "marked": "^4.3.0",
```
Comment (blackfalcon) [inline package.json:60]: Remove

File: package.json [hunk 12]
```diff
@@ -16,71 +16,69 @@
...
-    "sass": "^1.53.0",
-    "semantic-release": "^19.0.3",
-    "sinon": "^14.0.0",
-    "stylelint": "^14.9.1",
-    "stylelint-config-idiomatic-order": "^8.1.0",
-    "stylelint-config-standard": "^26.0.0",
-    "stylelint-config-standard-scss": "^4.0.0",
-    "stylelint-order": "^5.0.0",
-    "stylelint-scss": "^4.2.0",
+    "prismjs": "^1.29.0",
```
Comment (blackfalcon) [inline package.json:67]: Remove

File: package.json [hunk 13]
```diff
@@ -16,71 +16,69 @@
...
+    "markdown-magic": "^2.6.1",
+    "marked": "^4.3.0",
+    "nodemon": "^2.0.22",
     "npm-run-all": "^4.1.5",
-    "postcss": "^8.4.14",
-    "postcss-discard-comments": "^5.1.2",
+    "postcss": "^8.4.23",
+    "postcss-discard-comments": "^6.0.0",
     "postcss-remove-rules": "^1.0.0",
     "postcss-selector-replace": "^1.0.2",
```
Comment (blackfalcon) [inline package.json:66]: Remove

File: package.json [hunk 14]
```diff
@@ -16,71 +16,69 @@
...
-    "stylelint": "^14.9.1",
-    "stylelint-config-idiomatic-order": "^8.1.0",
-    "stylelint-config-standard": "^26.0.0",
-    "stylelint-config-standard-scss": "^4.0.0",
-    "stylelint-order": "^5.0.0",
-    "stylelint-scss": "^4.2.0",
+    "prismjs": "^1.29.0",
+    "rollup": "^3.21.0",
+    "rollup-plugin-serve": "^2.0.2",
+    "sass": "^1.62.1",
```
Comment (blackfalcon) [inline package.json:70]: Remove

File: index.js (hunk 7 above)
Comment (blackfalcon) [reply index.js:15]: Don't forget to update the following file/line as part of this update

https://github.com/AlaskaAirlines/auro-flight/blob/fajarapri/node18update/.npmignore#L21

**Review by blackfalcon — CHANGES_REQUESTED** [review]

File: package.json (hunk 12 above)
Comment (fajar-apri-alaska) [reply package.json:67]: wait, but then we need to revert changes in [here](https://github.com/AlaskaAirlines/auro-flight/pull/64/commits/037ffee53935530dcf6865a8e3fc649a80d13515), no?

File: demo/index.html (hunk 6 above)
Comment (fajar-apri-alaska) [reply demo/index.html:35]: right, it is already imported in each of the partial demo files.

File: package.json (hunk 13 above)
Comment (fajar-apri-alaska) [reply package.json:66]: Above dependencies are needed for scripts/postCss.js, right?

File: package.json (hunk 14 above)
Comment (fajar-apri-alaska) [reply package.json:70]: Above dependencies are needed by `build:sass` package.json script command, no?

Comment (blackfalcon) [reply package.json:70]: No. Sass is a dependency of wc-sass-render. It's included in its bundle.

**Review by blackfalcon — CHANGES_REQUESTED** [review]

File: package.json (hunk 12 above)
Comment (fajar-apri-alaska) [reply package.json:67]: Righty, my bad.

**Comment by blackfalcon** [comment]
This PR is BLOCKED. See the following test errors. 

https://github.com/AlaskaAirlines/auro-flight/actions/runs/4886549774/jobs/8722032969#step:5:99
https://github.com/AlaskaAirlines/auro-flight/actions/runs/4886549774/jobs/8722032969#step:5:108

The auro-flightline element must be updated to Lit 2.0 before this update can be completed.

File: src/auro-flight-main.js [hunk 15]
```diff
@@ -189,8 +189,7 @@ class AuroFlightMain extends LitElement {
   }
 }
 
-/* istanbul ignore else */
-// define the name of the custom component
+/* eslint max-statements: ["error", 11] */
```
Comment (fajar-apri-alaska) [inline src/auro-flight-main.js:192]: Hi @blackfalcon, do I also need to refactor this file to achieve this eslint requirement?

Comment (fajar-apri-alaska) [reply src/auro-flight-main.js]: For now, I'm setting this rule to 18 in this [commit](https://github.com/AlaskaAirlines/auro-flight/pull/64/commits/d0cb5d821be0dd0d87840eba0f4b4a63105c5021) for related file only.
Also, please advise if this is not needed. Thanks.

**Review by blackfalcon — DISMISSED** [review]

**Review by blackfalcon — APPROVED** [review]
